docs/how-certificate-issuing-works.md

```diff
@@ -0,0 +1,70 @@
+# Certificate Issuing Overview
+
+This document walks through the process of issuing a code signing certificate
+from start to finish. This is a great entry point to understanding how Fulcio
+works if you're interesting in contributing to the project or just want to
+learn more about whats happening under the hood.
+
+## 1 | Certificate Request Input
+
+To begin, the client submits a certificate request to Fulcio.
+
+![Certificate request diagram](img/certificate-request.png)
+
+The certificate request contains three items:
+
+- An OIDC identity token. This is a JWT containing information about the
+  principal (identity of our client), the issuer (who verified this identity?
+Google, Github etc) and some other metadata.
+- The public key. This is the public half of an asymmetric key-pair generated
+  by the client. The public key will be embedded in the final x509 certificate
+if this request is successful
+- A challenge. This challenge proves the client is in possession of the private
+  key that corresponds to the public key provided. The challenge created by
+signing the subject portion of the OIDC ID token
+
+## 2 | Authentication
+
+The first step in processing the certificate request is to authenticate the
+OIDC ID token.
+
+![OIDC Authentication diagram](img/authenticate-token.png)
+
+To authorize the token Fulcio must:
+
+- Use the issuer claim from the token find the issuers OIDC discovery endpoint
+- Download the issuers signing keys from the discovery endpoint
+- Verify the ID token signature
+
+## 3 | Verifying the challenge
+
+Once the client has been authenticated, the next step is to verify the client
+is in possession of the private key of the public key they’ve submitted. To do
+this we must verify the challenge. This is simply a signature of the `sub`
+claim so we verify the signature using the public key supplied.
```

Comment:
I guess you could mention here that the challenge uses the email as the element that is signed.

Reply:
The subject isn't always the email right? It can be the workflow URL or SPIFFE ID as well as I understand it. Thay is why I kept it general as the subject in the ID token, but perhaps I can enumerate the different things that can be?

Reply:
fair point